muikku-elastic-search/src/main/java/fi/otavanopisto/muikku/plugins/search/ElasticSearchProvider.java

```diff
@@ -166,8 +168,9 @@ public SearchResult searchUsers(String text, String[] textFields, Collection<Env
       if (startedStudiesBefore != null ) {
         query.must(rangeQuery("studyStartDate").lt((long) startedStudiesBefore.getTime() / 1000));
       }
-      
-      if(studyTimeEndsBefore != null) {
+
+      if (studyTimeEndsBefore != null) {
+        query.must(rangeQuery("studyTimeEnd").gte(now));
```

Comment:
Ok I guess it is in use in the notifier but the function declaration isn't necessarily explicit about the functionality (I would guess it returns all that have the studytimeend < specified regardless of current date).